pods-framework/pods
# Add setting to turn Pods into types-only mode which disables groups/fields hooks and screens

## Description

Types-only mode is geared towards people who want to use Pods for content types but prefer to use another custom field plugin. Types-only mode will be more performant and have reduced overhead on many areas as a result of no longer running hooks/logic checks on meta handling as well.

## Testing instructions



1. Go to Pods Admin > Settings
2. Enable / Disable Custom Fields handling
3. See the Groups / Fields no longer showing in Pods Admin > Edit Pods and the Fields tab is removed from the Edit Pod screen

## Screenshots / screencast

![Screen Shot 2022-02-05 at 6 09 01 PM](https://user-images.githubusercontent.com/709662/152663199-16e699a9-677d-499e-a137-b15820010d9a.png)

## Changelog text for these changes

> Added: New Types-only mode introduced in the Pods Admin > Settings screen for those wanting to use Pods only for content types. This mode improves performance and reduces hooks/logic where fields are concerned. #TBD (@sc0ttkclark, @zrothauser)

## PR checklist


## Remaining Tasks

- [x] Store the boolean option for podsAdminConfig.global.showFields in the DFV app for reference (@zrothauser)
- [x] If showFields is off, do not add "Fields" tab (@zrothauser)
- [x] If showFields is off, do not add Groups/Fields management to screen (@zrothauser)
- [x] If showFields is off, do not send Groups/Fields data to the REST API (@zrothauser)
